### PR TITLE
[SPARK-32907][ML] adaptively blockify instances - revert blockify gmm

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
revert blockify gmm

### Why are the changes needed?
@WeichenXu123  and I thought we should use memory size instead of number of rows to blockify instance; then if a buffer's size is large and determined by number of rows, we should discard it.
In GMM, we found that the pre-allocated memory maybe too large and should be discarded:
```
@transient private lazy val auxiliaryPDFMat = DenseMatrix.zeros(blockSize, numFeatures)
```
We had some offline discuss and thought it is better to revert blockify GMM.

### Does this PR introduce _any_ user-facing change?
blockSize added in master branch will be removed

### How was this patch tested?
existing testsuites

### CODE REVIEW

File: python/pyspark/ml/clustering.py [hunk 1]
```diff
@@ -432,13 +429,6 @@ def setAggregationDepth(self, value):
         """
         return self._set(aggregationDepth=value)
 
-    @since("3.1.0")
-    def setBlockSize(self, value):
```
Comment (HyukjinKwon) [inline python/pyspark/ml/clustering.py:436]: @zero323 FYI

**Review by srowen — COMMENTED** [review]
Seems OK if this is just reverting something that went into 3.1.

**Comment by srowen** [comment]
Feel free to merge

**Comment by WeichenXu123** [comment]
Jenkins retest this

**Comment by WeichenXu123** [comment]
Do we need to revert the change https://github.com/apache/spark/pull/27473/files#r493100573 and https://github.com/apache/spark/pull/27473/files#r493100907

**Comment by zhengruifeng** [comment]
@WeichenXu123  Good catch! Thanks

**Review by WeichenXu123 — APPROVED** [review]

**Comment by zhengruifeng** [comment]
Merged to master. Thanks all for reviewing!

**Pull request closed without merging**